core/common-util/src/test/java/uk/gov/gchq/gaffer/commonutil/CommonTimeUtilTest.java

```diff
@@ -37,12 +39,14 @@ public void shouldCorrectlyPlaceTimestampsIntoBuckets() {
         final OffsetDateTime offsetDateTime = OffsetDateTime.parse("2000-01-01T12:34:56.789Z");
         final long time = offsetDateTime.toInstant().toEpochMilli();
 
+        assertEquals(parse("2000-01-01T12:34:56.789Z").toInstant().toEpochMilli(), timeToBucket(time, MILLISECOND));
         assertEquals(parse("2000-01-01T12:34:56.000Z").toInstant().toEpochMilli(), timeToBucket(time, SECOND));
         assertEquals(parse("2000-01-01T12:34:00.000Z").toInstant().toEpochMilli(), timeToBucket(time, MINUTE));
         assertEquals(parse("2000-01-01T12:00:00.000Z").toInstant().toEpochMilli(), timeToBucket(time, HOUR));
         assertEquals(parse("2000-01-01T00:00:00.000Z").toInstant().toEpochMilli(), timeToBucket(time, DAY));
         assertEquals(parse("1999-12-27T00:00:00.000Z").toInstant().toEpochMilli(), timeToBucket(time, WEEK));
         assertEquals(parse("2000-01-01T00:00:00.000Z").toInstant().toEpochMilli(), timeToBucket(time, MONTH));
+        assertEquals(parse("2000-01-01T00:00:00.000Z").toInstant().toEpochMilli(), timeToBucket(time, YEAR));
```

Comment:
Perhaps you could update this test to use a date like 2000-02-03 and check it is truncated to 2000-01-01?